### ISSUE TITLE
[Go to definition] Should work on signature files nicely

### ISSUE THREAD

**Issue by vasily-kirichenko** [fsprojects/VisualFSharpPowerTools#1351]
* F12 should always jump to implementation (currently it jumps to function implementation and to type signature (fsi))
* F12 on a definition should switch between FS and FSI files circularly

**Comment by vasily-kirichenko** [comment]
I think it's easy to implement because FCS seems to return all symbol uses and `FSharpSymbolUse.IsFromDeclaration` is properly set to `true` for declaration in both fsi and fs files:

![image](https://cloud.githubusercontent.com/assets/873919/13965849/0d9aa450-f081-11e5-8bd3-4052b2d20646.png)

**Comment by vasily-kirichenko** [comment]
You are right :)

OK, what about this:

* If we in an FSI file, we always jump to FSI file (if available)
* If we in an FS file, we always jump to FS file
* If we are on a definition, then we jump FSI <-> FS recursively

**Comment by dungpa** [comment]
Sounds reasonable.

Implementing this means overriding Visual F# tool's F12 completely. 
Are you fine with that? An alternative is to have a new command (e.g. Alt Shift F12) to switch between declarations on implementation and signature files.

**Comment by vasily-kirichenko** [comment]
I think "F12 for everything" approach is more straightforward and easier to use.

**Comment by vasily-kirichenko** [comment]
OK, I found a bug in FCS, so cannot proceed with the feature until it's fixed.

**Comment by dungpa** [comment]
You're right. I made an attempt here https://github.com/dungpa/PowerTools/tree/f12, but it obviously doesn't work since ImplementationRange, DeclarationRange often point the same location.

**Comment by vasily-kirichenko** [comment]
@dungpa I've made similar changes here https://github.com/fsprojects/VisualFSharpPowerTools/pull/1352 :)
